Source/Scene/ModelExperimental/ModelExperimental.js

```diff
@@ -0,0 +1,369 @@
+import clone from "../../Core/clone.js";
+import defined from "../../Core/defined.js";
+import defaultValue from "../../Core/defaultValue.js";
+import DeveloperError from "../../Core/DeveloperError.js";
+import GltfLoader from "../GltfLoader.js";
+import ModelExperimentalUtility from "./ModelExperimentalUtility.js";
+import ModelExperimentalSceneGraph from "./ModelExperimentalSceneGraph.js";
+import Resource from "../../Core/Resource.js";
+import when from "../../ThirdParty/when.js";
+import destroyObject from "../../Core/destroyObject.js";
+
+/**
+ * A 3D model based on glTF, the runtime asset format for WebGL. This is
+ * a new architecture that is more decoupled than the older {@link Model}.
+ *
+ * This class is still experimental. glTF features that are core to 3D Tiles
+ * are supported, but other features such as animation are not yet supported.
+ *
+ * @constructor
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {String|Resource|ArrayBuffer|Uint8Array} [options.gltf] A Resource/URL to a glTF/glb file or a binary glTF buffer.
+ * @param {Resource|String} [options.basePath=''] The base path that paths in the glTF JSON are relative to.
+ * @param {Matrix4} [options.modelMatrix=Matrix4.IDENTITY]  The 4x4 transformation matrix that transforms the model from model to world coordinates.
+ * @param {Boolean} [options.incrementallyLoadTextures=true] Determine if textures may continue to stream in after the model is loaded.
+ * @param {Boolean} [options.releaseGltfJson=false] When true, the glTF JSON is released once the glTF is loaded. This is is especially useful for cases like 3D Tiles, where each .gltf model is unique and caching the glTF JSON is not effective.
+ * @param {Boolean} [options.debugShowBoundingVolume=false] For debugging only. Draws the bounding sphere for each draw command in the model.
```

Comment:
@lilleyse just to be clear, these constructors were written under the assumption that we want to stay as close to the `Model` constructor parameters to avoid big breaking changes when someday we replace `Model` with `ModelExperimental`?

Reply:
even if you kept `fromGltf`, the thing is the old model's constructor is still [part of the public API](https://cesium.com/learn/cesiumjs/ref-doc/Model.html?classFilter=model). What do you suggest we do here?

Reply:
This would be an acceptable breaking change. The Model constructor isn't used that frequently and there should be a pretty clear path for using `Model.fromGltf` instead.

Reply:
@ebogo1 For your awareness, with the new `ModelExperimental`, the constructor should no longer be directly called. In `Gltf3DTileContent`, we'll be using `fromGltf` when the experimental flag is set and the old `Model` constructor otherwise.